did_peer/src/numalgos/numalgo2/generate/mod.rs

```diff
@@ -0,0 +1,17 @@
+mod helpers;
+
+use did_doc::schema::did_doc::DidDocument;
+use did_doc_sov::extra_fields::ExtraFieldsSov;
+
+use crate::{error::DidPeerError, peer_did::PeerDid};
+
+use self::helpers::{append_encoded_key_segments, append_encoded_service_segment};
+
+pub fn generate_numalgo2(did_document: DidDocument<ExtraFieldsSov>) -> Result<PeerDid, DidPeerError> {
+    let mut did = String::from("did:peer:2");
+
+    did = append_encoded_key_segments(did, &did_document)?;
+    did = append_encoded_service_segment(did, &did_document)?;
+
+    PeerDid::parse(did)
```

Comment:
Perhaps this could be made more efficiently if the `PeerDid::parse_numalgo()` and `PeerDid::parse_transform()` would process the unprefixed strings (or just the chars, actually) instead of processing the entire `Did` for which the prefix is known to be correct since it's being added here.

Reply:
I am sorry, I don't think this micro-optimization is worth it in the slightest. Also, from the perspective of `PeerDid`, the prefix is not known to be correct.